gadget/install/content.go

```diff
@@ -110,5 +111,22 @@ func writeFilesystemContent(laidOut *gadget.LaidOutStructure, fsDevice string, o
 		return fmt.Errorf("cannot create filesystem image: %v", err)
 	}
 
+	// For data partition, copy drivers tree if it has been created, so
+	// kernel drivers are available on first boot of installed system.
+	if laidOut.Role() == gadget.SystemData {
+		src := dirs.SnapKernelDriversTreesDirUnder(dirs.GlobalRootDir)
+		exists, isDir, _ := osutil.DirExists(src)
+		if exists && isDir {
+			dst := dirs.SnapKernelDriversTreesDirUnder(filepath.Join(mountpoint, "system-data"))
+			logger.Noticef("copying drivers tree to %s", laidOut.Name())
+			if err := os.MkdirAll(dst, 0755); err != nil {
+				return err
+			}
+			if output, stderr, err := osutil.RunSplitOutput("cp", "-aT", src, dst); err != nil {
```

Comment:
don't we have something in osutil that encapsulate this?

Reply:
there's some code in snapstate backend which uses `osutil.CopyFile(..., osutil.CopyFlagPreserveAll)` which in turn ends up using `cp -av`

Reply:
The new approach does not copy file anymore